drivers/sensor/ina23x/ina23x.c

```diff
@@ -0,0 +1,363 @@
+	case SENSOR_ATTR_CONFIGURATION:
+		ret = ina23x_reg_read(dev, INA23X_REG_CONFIG, &data);
+		if (ret < 0) {
+			return ret;
+		}
+		break;
+	case SENSOR_ATTR_CALIBRATION:
+		ret = ina23x_reg_read(dev, INA23X_REG_CALIB, &data);
+		if (ret < 0) {
+			return ret;
+		}
+		break;
+	case SENSOR_ATTR_MASK:
+		ret = ina23x_reg_read(dev, INA23X_REG_MASK, &data);
+		if (ret < 0) {
+			return ret;
+		}
+		break;
+	case SENSOR_ATTR_ALERT:
+		ret = ina23x_reg_read(dev, INA23X_REG_ALERT, &data);
+		if (ret < 0) {
+			return ret;
+		}
+		break;
+	default:
+		LOG_ERR("INA23X attribute not supported.");
+		return -ENOTSUP;
+	}
+
+	val->val1 = data;
+	val->val2 = 0;
+
+	return 0;
+}
+
+/**
+ * @brief Initialize the INA23x
+ *
+ * @return 0 for success
+ * @return -EINVAL on error
+ */
+static int ina23x_init(const struct device *dev)
+{
+	struct ina23x_data *ina23x = dev->data;
+	const struct ina23x_config *const config = dev->config;
+	uint16_t cal;
+	int ret;
+
+	ina23x->i2c = device_get_binding(config->bus_name);
+	if (ina23x->i2c == NULL) {
+		LOG_ERR("Failed to get pointer to %s device!",
+			config->bus_name);
+		return -EINVAL;
+	}
+
+	ret = ina23x_reg_write(dev, INA23X_REG_CONFIG, config->config);
+	if (ret < 0) {
+		LOG_ERR("Failed to write configuration register!");
+		return ret;
+	}
+
+	cal = INA23X_CALIB(config->current_lsb, config->rshunt);
+
+	ret = ina23x_reg_write(dev, INA23X_REG_CALIB, cal);
+	if (ret < 0) {
+		LOG_ERR("Failed to write calibration register!");
+		return ret;
+	}
+
+#ifdef CONFIG_INA23X_TRIGGER
+	if (config->trig_enabled) {
+		ret = ina23x_trigger_mode_init(dev);
+		if (ret < 0) {
+			LOG_ERR("Failed to init trigger mode\n");
+			return ret;
+		}
+
+		ret = ina23x_reg_write(dev, INA23X_REG_ALERT,
+				       config->alert_limit);
+		if (ret < 0) {
+			LOG_ERR("Failed to write alert register!");
+			return ret;
+		}
+
+		ret = ina23x_reg_write(dev, INA23X_REG_MASK, config->mask);
+		if (ret < 0) {
+			LOG_ERR("Failed to write mask register!");
+			return ret;
+		}
+	}
+#endif  /* CONFIG_INA23X_TRIGGER */
+
+	return 0;
+}
+
+static const struct sensor_driver_api ina23x_driver_api = {
+	.attr_set = ina23x_attr_set,
+	.attr_get = ina23x_attr_get,
+#ifdef CONFIG_INA23X_TRIGGER
+	.trigger_set = ina23x_trigger_set,
+#endif
+	.sample_fetch = ina23x_sample_fetch,
+	.channel_get = ina23x_channel_get,
+};
+
+BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) > 0,
+	     "No compatible ina23x instances found");
+
+#ifdef CONFIG_INA23X_TRIGGER
+#define INA23X_CFG_IRQ(inst)				\
+	.trig_enabled = true,				\
+	.mask = DT_INST_PROP(inst, mask),		\
+	.alert_limit = DT_INST_PROP(inst, alert_limit),	\
+	.gpio_alert = GPIO_DT_SPEC_INST_GET(inst, irq_gpios)
+#else
+#define INA23X_CFG_IRQ(inst)
+#endif /* CONFIG_INA23X_TRIGGER */
+
+#define INA23X_DRIVER_INIT(inst)				    \
+	static struct ina23x_data drv_data_##inst;		    \
+	static const struct ina23x_config drv_config_##inst = {	    \
+		.bus_name = DT_INST_BUS_LABEL(inst),		    \
```

Comment:
missed that one, better use `bus = DEVICE_DT_GET(DT_INST_BUS(inst))`, see e.g. https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/sensor/lis2mdl/lis2mdl.c#L593